src/ray/gcs/gcs_server/gcs_task_manager.h

```diff
@@ -174,6 +238,13 @@ class GcsTaskManager : public rpc::TaskInfoHandler {
     absl::flat_hash_map<JobID, absl::flat_hash_set<TaskAttempt>>
         job_to_task_attempt_index_;
 
+    /// Secondary index from child task id to parent task id.
+    absl::flat_hash_map<TaskID, TaskID> child_to_parent_task_index_;
```

Comment:
Isn't this unnecessary since parent id can be looked up as part of the task info struct?

Reply:
Ah right - I think looking up the task info is probably better, comparing to maintaining this.